feat(android): Add Jetpack Compose automatic UI transaction docs (#5893)
getsentry/sentry-docs: src/platforms/android/common/performance/instrumentation/automatic-instrumentation.mdx

### User Interaction Instrumentation

> The UI instrumentation, once enabled, captures transactions for a set of different user interactions, which include clicks, scrolls and swipes. User interaction instrumentation is available for both classic Android Views as well as Jetpack Compose.

Reviewer: ```suggestion
The UI instrumentation, once enabled, captures transactions for a set of different user interactions, which include clicks, scrolls, and swipes. User interaction instrumentation is available for both classic Android Views as well as Jetpack Compose.
```

> The UI instrumentation, once enabled, captures transactions for a set of different user interactions, which include clicks, scrolls and swipes. User interaction instrumentation is available for both classic Android Views as well as Jetpack Compose UI.

Reviewer: ```suggestion
The UI instrumentation, once enabled, captures transactions for a set of different user interactions, which include clicks, scrolls and swipes. User interaction instrumentation is available for both classic Android Views as well as the Jetpack Compose UI.
```

> The UI instrumentation, once enabled, captures transactions for a set of different user interactions, which include clicks, scrolls and swipes. User interaction instrumentation is available for both classic Android Views as well as Jetpack Compose UI.

Author: ```suggestion
The UI instrumentation, once enabled, captures transactions for a set of different user interactions, which include clicks, scrolls and swipes. User interaction instrumentation is available for both classic Android Views as well as Jetpack Compose.
```

This feature is disabled by default, but you can enable it by setting:

```xml {filename:AndroidManifest.xml}
<application>
    <meta-data android:name="io.sentry.traces.user-interaction.enable" android:value="true" />
</application>
```

> #### User Interaction Instrumentation for Classic Android View System

Reviewer: ```suggestion
#### User Interaction Instrumentation for Classic Android View System

```

> #### User Interaction Instrumentation for classic Android View System

Reviewer: ```suggestion
#### User Interaction Instrumentation for Classic Android View System
```

_(New in version 6.0.0)_

>  The SDK composes the transaction name out of the host `Activity` and the `id` of the view, which captured the user interaction; for example, `LoginActivity.login_button`. The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll` or `swipe`).

Reviewer: ```suggestion
 The SDK composes the transaction name out of the host `Activity` and the `id` of the view that captured the user interaction — for example, for `LoginActivity.login_button`. The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll`, or `swipe`).
```

<Note>

If the view doesn't have the `id` assigned, the transaction won't be captured because it can't be uniquely identified otherwise.

</Note>

> #### User Interaction Instrumentation for Jetpack Compose

Reviewer: ```suggestion
#### User Interaction Instrumentation for Jetpack Compose

```

_(New in version 6.10.0)_

>  The SDK composes the transaction name out of the host `Activity` and the `tag` set via `Modifier.testTag("<tag>")` of the Composable. For example, `LoginActivity.login_button`. The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll` or `swipe`).

Reviewer: ```suggestion
The SDK composes the transaction name out of the host `Activity` and the `tag` set by way of `Modifier.testTag("<tag>")` of the Composable (for example, `LoginActivity.login_button`). The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll`, or `swipe`).
```

> The SDK composes the transaction name out of the host `Activity` and the `tag` set by way of `Modifier.testTag("<tag>")` of the Composable (for example, `LoginActivity.login_button`). The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll`, or `swipe`).

Author: Good point! @romtsn and I discussed about this as well.`.testTag()` is used for writing Jetpack Compose UI tests to identify an element within the view hierarchy. So the use case is quite similar to ours.
I see two advantages:
* it will work out of the box for any existing Composables
* there's no need for an extra API
* If we would introduce our own modifier, the users would probably end up setting the same value there as well:
````
Button(
    modifier = Modifier
      .testTag("button_login")
      .sentryTag("button_login") {
    ...
}
````
But yeah I see your point, it still could be confusing. Maybe let's wait for some user feedback and add our own API if necessary?

> The SDK composes the transaction name out of the host `Activity` and the `tag` set by way of `Modifier.testTag("<tag>")` of the Composable (for example, `LoginActivity.login_button`). The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll`, or `swipe`).

Reviewer 2: yep - our ultimate goal would also be to automatically generate the transaction name through SAGP (either bytecode manipulation or kotlin compiler plugin), so long-term users will only need this as a fallback in rare cases, so I think it's a good middle-ground

> The SDK composes the transaction name out of the host `Activity` and the `tag` set by way of `Modifier.testTag("<tag>")` of the Composable (for example, `LoginActivity.login_button`). The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll`, or `swipe`).

Reviewer 3: What about using `testTag` as a fallback if `sentryTag` is not set? So if you are already using `testTag` you are fine, because you don't have to set it twice. If you use both, `sentryTag` wins. Finally, we recommend using `sentryTag`. I'm also fine with not adding `sentryTag` in the first iteration. It's always easier to add APIs than to remove them.

> The SDK composes the transaction name out of the host `Activity` and the `tag` set by way of `Modifier.testTag("<tag>")` of the Composable (for example, `LoginActivity.login_button`). The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll`, or `swipe`).

Reviewer: ```suggestion
The SDK composes the transaction name out of the host `Activity` and the `tag` set by way of the `Modifier.testTag("<tag>")` of the `Composable` (for example, `LoginActivity.login_button`). The transaction operation is set to `ui.action` plus the interaction type (one of `click`, `scroll`, or `swipe`).
```
